ebl/tests/fragmentarium/test_fragment_repository.py

```
@@ -105,6 +105,42 @@ def test_find_interesting(fragment_repository):
     ) == [interesting_fragment]
 
 
+def test_folio_pager_exception(fragment_repository):
+    query = "1841-07-26, 57"
+    with pytest.raises(NotFoundError):
+        list(fragment_repository.query_next_and_previous_fragment(query).values())
+
+
+FRAGMENTS = ["1841-07-26, 54", "1841-07-26, 57", "1841-07-26, 63"]
+
+
+@pytest.mark.parametrize(
+    "query,  existing,expected",
+    [
+        ("1841-07-26, 57", FRAGMENTS, ["1841-07-26, 54", "1841-07-26, 63"]),
+        ("1841-07-26, 63", FRAGMENTS, ["1841-07-26, 57", "1841-07-26, 54"]),
+        ("1841-07-26, 54", FRAGMENTS, ["1841-07-26, 63", "1841-07-26, 57"]),
+        ("1841-07-26, 54", FRAGMENTS[:2], ["1841-07-26, 57", "1841-07-26, 57"]),
+    ],
+)
+def test_query_next_and_previous_fragment(
+    query, existing, expected, fragment_repository
+):
+    fragments = [FragmentFactory.build(number=x) for x in existing]
+
+    for fragment in fragments:
+        fragment_repository.create(fragment)
+
+    results = list(fragment_repository.query_next_and_previous_fragment(query).values())
+    assert results == expected
+
+
+def test_query_next_and_previous_fragment_exception(fragment_repository):
+    query = "1841-07-26, 57"
+    with pytest.raises(NotFoundError):
+        list(fragment_repository.query_next_and_previous_fragment(query).values())
```

list and .values() are not needed. Making the call should raise the exception.